### PR TITLE
[fix #137] check sink url

### LINKED ISSUE #133: cdc: Add sorter back to pipeline
### 1. Describe the bug

The sorter of pipeline was removed in early cdc [PR](https://github.com/tikv/migration/pull/74/files#diff-571e6402cf7475dcd07a4193de26b908581ff481d6f698a43b9c4ab17af21933).
And that time we thought that as timestamps of entries in a region are always non-decreasing, we don't need the sorter.

But now I find that actually the processor is not processing only one region, but multiple regions at a time. And now the timestamps are not in order. This will make the resolved-ts logic wrong.
See [here](https://github.com/tikv/migration/blob/48c96505cdcca73dceb1aa1476597f62d3661dc6/cdc/cdc/sink/buffer_sink.go#L129):
```go
	keyspanID, resolvedTs := batch[i].keyspanID, batch[i].resolvedTs
	rawKVEntries := b.buffer[keyspanID]

	i := sort.Search(len(rawKVEntries), func(i int) bool {
		return rawKVEntries[i].CRTs > resolvedTs
	})
	if i == 0 {
		continue
	}
	state.metricTotalRows.Add(float64(i))

	err := b.Sink.EmitChangedEvents(ctx, rawKVEntries...)
```

If the CRTs of entries are not in order, we would not get all the expected entries that earlier than `resolvedTs`.

Although we can do a full search here, but it would be inefficiency. As the advancing of `resolvedTs` would be slow, we will scan again and again.

We can implement the sorter using merge sort, to utilize the feature of non-decreasing.

Please also note that, it's possible that we would break the order of entires timestamp in the same region later. See https://github.com/tikv/tikv/issues/12020.

### 5. What is your migration tool and TiKV version? (Required)

- TiKV:
- TiKV CDC:
- TiKV BR:
- TiKV Online Bulk Load:

### LINKED ISSUE #137: cdc cli command fail to detect wrong format of sink-uri
### 1. Describe the bug

The sink-uri shoudl not have the `http://` prefix, but cli fail to detect or correct this automatically.

image

cli create changefeed successfully, but pd client is continuously failing in backgroud.

_[WARN] [base_client.go:251] ["[pd] failed to get cluster id"] [url=http://http:] [error="[PD:client:ErrClientGetMember]error:rpc error: code = Unavailable desc = connection error: desc = \"transport: Error while dialing dial tcp: lookup http on 172.16.4.1:53: no such host\" target:http: status:TRANSIENT_FAILURE: error:rpc error: code = Unavailable desc = connection error: desc = \"transport: Error while dialing dial tcp: lookup http on 172.16.4.1:53: no such host\" target:http: status:TRANSIENT_FAILURE"]_

### 2. Minimal reproduce step (Required)

bin/cdc cli changefeed create --pd http://172.16.5.32:2379 --changefeed-id feed016 --sort-engine unified --sink-uri='tikv://http://172.16.5.32:39244'

### 4. What did you expect to see? (Required)

cdc cli command should detect the wrong format and fails, or correct the format automatically. 

### 5. What is your migration tool and TiKV version? (Required)

- TiKV CDC:

### PR DESCRIPTION
**Pull request by zeminzhou**
Signed-off-by: zeminzhou <zhouzemin@pingcap.com>

### What problem does this PR solve?
check the format of sink url, when create changefeed.

Issue Number: close #137

Problem Description: **TBD**
see issue #137

### What is changed and how does it work?
add check for the format of sink url.

### Code changes

- Has exported function/method change

### Check List for Tests

This PR has been tested by at least one of the following methods:
- Unit test

### CODE REVIEW

**Review by haojinming — DISMISSED** [review]
LGTM

File: cdc/pkg/cmd/cli/cli_changefeed_create.go [hunk 1]
```diff
@@ -180,14 +180,14 @@ func (o *createChangefeedOptions) completeCfg(ctx context.Context, cmd *cobra.Co
 
 	cfg := config.GetDefaultReplicaConfig()
 	if len(o.commonChangefeedOptions.configFile) > 0 {
-		if err := o.commonChangefeedOptions.strictDecodeConfig("TiCDC changefeed", cfg); err != nil {
+		if err := o.commonChangefeedOptions.strictDecodeConfig("TiKVCDC changefeed", cfg); err != nil {
```
Comment (pingyu) [inline cdc/pkg/cmd/cli/cli_changefeed_create.go:183]:
```suggestion
		if err := o.commonChangefeedOptions.strictDecodeConfig("TiKV-CDC changefeed", cfg); err != nil {
```

File: cdc/pkg/cmd/cli/cli_changefeed_create.go [hunk 2]
```diff
@@ -180,14 +180,14 @@ func (o *createChangefeedOptions) completeCfg(ctx context.Context, cmd *cobra.Co
...
 			return err
 		}
 	}
 
 	if !cdcClusterVer.ShouldEnableOldValueByDefault() {
 		cfg.EnableOldValue = false
-		log.Warn("The TiCDC cluster is built from an older version, disabling old value by default.",
+		log.Warn("The TiKVCDC cluster is built from an older version, disabling old value by default.",
```
Comment (pingyu) [inline cdc/pkg/cmd/cli/cli_changefeed_create.go:190]:
```suggestion
		log.Warn("The TiKV-CDC cluster is built from an older version, disabling old value by default.",
```

**Review by pingyu — APPROVED** [review]
LGTM~

**Review by haojinming — APPROVED** [review]

**Pull request merged**